pkg/api/api.go

```diff
@@ -231,6 +231,11 @@ func (a *API) RegisterPurger(store *purger.DeleteStore, deleteRequestCancelPerio
 	a.RegisterRoute(a.cfg.LegacyHTTPPrefix+"/api/v1/admin/tsdb/cancel_delete_request", http.HandlerFunc(deleteRequestHandler.CancelDeleteRequestHandler), true, "PUT", "POST")
 }
 
+func (a *API) RegisterBlocksPurger(api *purger.BlocksPurgerAPI) {
+	a.RegisterRoute("/purger/delete_tenant", http.HandlerFunc(api.DeleteTenant), true, "POST")
+	a.RegisterRoute("/purger/delete_tenant_status", http.HandlerFunc(api.DeleteTenantStatus), true, "GET")
```

Comment:
**Thought:** It's not an official convention, however, requests that are scoped to a particular tenant universally have a `/api/v1` prefix. The only exception I know of is the Alertmanager UI which is a bit of an edge case. It might be worth following/codifying that convention in this PR. However, I can see an argument to be made that this is a bit different since it eliminates the tenant from existence.

```suggestion
	a.RegisterRoute("/api/v1/delete_tenant", http.HandlerFunc(api.DeleteTenant), true, "POST")
	a.RegisterRoute("/api/v1/delete_tenant_status", http.HandlerFunc(api.DeleteTenantStatus), true, "GET")
```

Reply:
I am also conflicted about this. On one hand, this is part of API, and we will keep it stable once development has finished.

On the other hand, it's part of "admin" API (eg. like /shutdown), and don't want to send wrong signal by putting it under `/api/v1` prefix. Someone may accidentally expose entire `/api/v1/*` to the users, which in case of `delete_tenant` would be pretty bad.

I'd suggest to keep it as is for now, we can always decide on supporting different path in the future.